tests/performance_tests/performance_test_basic.py

```diff
@@ -0,0 +1,459 @@
+                testSuccessful,
+                self.testHelperConfig._killEosInstances,
+                self.testHelperConfig._killWallet,
+                self.testHelperConfig.keepLogs,
+                self.testHelperConfig.killAll,
+                self.testHelperConfig.dumpErrorDetails
+                )
+
+            if not self.delPerfLogs:
+                self.captureLowLevelArtifacts()
+
+            if not completedRun:
+                os.system("pkill trx_generator")
+                print("Test run cancelled early via SIGINT")
+
+            if self.delPerfLogs:
+                print(f"Cleaning up logs directory: {self.testTimeStampDirPath}")
+                self.testDirsCleanup(self.delReport)
+
+            return testSuccessful
+
+def parseArgs():
+    appArgs=AppArgs()
+    appArgs.add(flag="--target-tps", type=int, help="The target transfers per second to send during test", default=8000)
+    appArgs.add(flag="--tps-limit-per-generator", type=int, help="Maximum amount of transactions per second a single generator can have.", default=4000)
+    appArgs.add(flag="--test-duration-sec", type=int, help="The duration of transfer trx generation for the test in seconds", default=90)
+    appArgs.add(flag="--genesis", type=str, help="Path to genesis.json", default="tests/performance_tests/genesis.json")
+    appArgs.add(flag="--num-blocks-to-prune", type=int, help=("The number of potentially non-empty blocks, in addition to leading and trailing size 0 blocks, "
+                                                              "to prune from the beginning and end of the range of blocks of interest for evaluation."), default=2)
+    appArgs.add(flag="--signature-cpu-billable-pct", type=int, help="Percentage of actual signature recovery cpu to bill. Whole number percentages, e.g. 50 for 50%%", default=0)
+    appArgs.add(flag="--chain-state-db-size-mb", type=int, help="Maximum size (in MiB) of the chain state database", default=10*1024)
+    appArgs.add(flag="--disable-subjective-billing", type=bool, help="Disable subjective CPU billing for API/P2P transactions", default=True)
+    appArgs.add(flag="--last-block-time-offset-us", type=int, help="Offset of last block producing time in microseconds. Valid range 0 .. -block_time_interval.", default=0)
+    appArgs.add(flag="--produce-time-offset-us", type=int, help="Offset of non last block producing time in microseconds. Valid range 0 .. -block_time_interval.", default=0)
+    appArgs.add(flag="--cpu-effort-percent", type=int, help="Percentage of cpu block production time used to produce block. Whole number percentages, e.g. 80 for 80%%", default=100)
+    appArgs.add(flag="--last-block-cpu-effort-percent", type=int, help="Percentage of cpu block production time used to produce last block. Whole number percentages, e.g. 80 for 80%%", default=100)
+    appArgs.add(flag="--http-max-response-time-ms", type=int, help="Maximum time for processing a request, -1 for unlimited", default=990000)
```

Comment:
Is there any alternative to replicating many of nodeos configuration options in a second location like this? (newer PRs have even more than the ~10 in this PR). Seems like a sneaky maintenance overhead, and especially prone to breakage for options not used by default. 

A simple naive way to do this is to have a way to "forward" arguments to nodeos without the python code caring, For example instead of
```
perf.py --target-tps --test-duration-sec 60 --genesis ~/genesis.json --signature-cpu-billable-pct 50 --chain-state-db-size-mb 40
```
it becomes
```
perf.py --target-tps --test-duration-sec 60 -- --genesis ~/genesis.json --signature-cpu-billable-pct 50 --chain-state-db-size-mb 40
```
that is, everything after `--` is simply forwarded along to nodeos.

Unfortunately that won't work here due to the defaults for these options in this performance test not being the same as the defaults of nodeos.

Reply:
>Is there any alternative to replicating many of nodeos configuration options in a second location like this? (newer PRs have even more than the ~10 in this PR). Seems like a sneaky maintenance overhead, and especially prone to breakage for options not used by default.

Some thoughts here:

There are several alternatives that are being discussed down at the Regression Test Framework (`TestHarness`) level, where `TestHelper.py`, `Node.py` and `Cluster.py` boostrap the blockchain test environment.  The Performance Harness MVP was not intending to implement those during this effort, but will look to take advantage of them and integrate with them when they occur.

Currently in `Cluster.py` there is a catch-all argument `extraNodeosArgs` that functions similarly to your suggestion to pass on any command line provided extra arguments, not explicitly supported by a command line argument, directly to `nodeos`.  However, in the Performance Harness there is a desire to have finer grain manipulation of configurations between specific nodes to measure specific configurations.  We are trying to identify which arguments have performance impacts, and provide support directly for those.  We are also providing defaults for most config options to either mirror those in `nodeos` itself or those that have been tuned for performance due to empirical analysis.

As far as maintainability is concerned, there is a basic test being run in CI/CD, which should catch if any config items change at the `nodeos` level that need to bubble up into the regression test and performance frameworks, since all config parameters used by the performance test (even those defaulted) are being passed to nodeos, and would cause a test failure if no longer matching `nodeos` options.

An effort was just completed to group configuration options such that they would not have to be reproduced at each level of script in the Performance Harness.  The use of `ArgumentParser` `parents` and `groups` will help bubble component arguments up/down for reuse as well as provide groups for easier understanding of when to use each.  That work is currently sitting in a PR for Stage 2 development here: https://github.com/AntelopeIO/leap/pull/518.

Care was taken to make it easy to expand the supported set of `nodeos` configuration options.  Adding a new command line option to `nodeos` is as simple as adding the argument in the `ArgumentParser` and in the plugin args object (`PerformanceBasicTest.ClusterConfig.ExtraNodeosArgs.*PluginArgs`).